packages/core/test/reputationProof.test.ts

```diff
@@ -426,4 +426,95 @@ describe('Reputation proof', function () {
         checkSignals(signals, proof)
         userState.stop()
     })
+
+    it('should not prove the reputation exceeds bit constraint in circuits', async () => {
+        const accounts = await ethers.getSigners()
+        const attester = accounts[1]
+        const attesterId = BigInt(attester.address)
+        const id = new Identity()
+        const userState = await genUserState(
+            ethers.provider,
+            unirepContract.address,
+            id,
+            attesterId
+        )
+
+        const epoch = await userState.sync.loadCurrentEpoch()
+        {
+            const { publicSignals, proof } = await userState.genUserSignUpProof(
+                { epoch }
+            )
+            await unirepContract
+                .connect(attester)
+                .userSignUp(publicSignals, proof)
+                .then((t) => t.wait())
+        }
+        await userState.waitForSync()
+        const repLimit = BigInt(1) << BigInt(64)
+
+        const epochKey = genEpochKey(
+            id.secret,
+            attester.address,
+            epoch,
+            0,
+            chainId
+        )
+        const field = userState.sync.settings.sumFieldCount
+
+        // should be able to receive, because data field is 254-bits
+        await unirepContract
+            .connect(attester)
+            .attest(epochKey, epoch, field, repLimit)
+            .then((t) => t.wait())
+
+        // setting posRep to 5
+        await unirepContract
+            .connect(attester)
+            .attest(epochKey, epoch, 0, 5)
+            .then((t) => t.wait())
+
+        // setting negRep to 3
+        await unirepContract
+            .connect(attester)
+            .attest(epochKey, epoch, 1, 3)
+            .then((t) => t.wait())
```

Comment:
Just curious about if setting rep here is necessary or not?

Reply:
oh just accidentally copied here